apps/src/templates/BaseDialog.jsx

```diff
@@ -0,0 +1,148 @@
+// TODO: Temporary measure to make sure code-studio doesn't end up with two Reacts
+var React = window.React;
+
+/**
+ * BaseDialog
+ * A generic modal dialog that has an x-close in the upper right, and a
+ * semi-transparent backdrop. Can be closed by clicking the x, clicking the
+ * backdrop, or pressing esc.
+ */
+var BaseDialog = React.createClass({
+  propTypes: {
+    isOpen: React.PropTypes.bool,
+    handleClose: React.PropTypes.func,
+    uncloseable: React.PropTypes.bool,
+    hideBackdrop: React.PropTypes.bool,
+    useDeprecatedGlobalStyles: React.PropTypes.bool,
+  },
+
+  componentDidMount: function () {
+    this.focusDialog();
+  },
+
+  componentDidUpdate: function () {
+    this.focusDialog();
+  },
+
+  closeOnEscape: function (event) {
+    if (event.key === 'Escape') {
+      this.closeDialog();
+    }
+  },
+
+  closeDialog: function () {
+    if (!this.props.uncloseable && this.props.handleClose) {
+      this.props.handleClose();
+    }
+  },
+
+  focusDialog: function () {
+    if (this.props.isOpen) {
+      this.refs.dialog.focus();
+    }
+  },
+
+  render: function () {
+    if (!this.props.isOpen && !this.props.hideBackdrop) {
+      return <div></div>;
+    }
+
+    var bodyStyle;
+    if (this.props.hideBackdrop) {
+      bodyStyle = {
+        position: 'initial',
+        marginLeft: 0,
+      };
+    }
+
+    let modalClassNames = "modal";
+    let modalBodyClassNames = "modal-body";
+    let modalBackdropClassNames = "modal-backdrop";
+
+    if (this.props.useDeprecatedGlobalStyles) {
+      modalClassNames = "modal dash_modal in";
+      modalBodyClassNames = "modal-body dash_modal_body";
+      modalBackdropClassNames = "modal-backdrop in";
+    }
```

Comment:
It's worth searching the entire source tree for these (e.g. https://github.com/code-dot-org/code-dot-org/search?utf8=%E2%9C%93&q=modal-body) because while they most often show up in .haml, they also appear to show up in .jsx, .js, and UI tests, etc.  Not familiar with all the uses of these terms; in fact most of early modal stuff was up and running before I ever worked on dashboard.